Well, I tried to have `bp` as local, but make wasn't really about it. 
Pushing now with both bp and logbp as part of model type. 

```
[ 42%] Building Fortran object extern/fabm/models/pisces/src/CMakeFiles/fabm_models_pisces.dir/phytoplankton.F90.o
/home/arthur/Desktop/CODES/SEAMLESS/fabm-pisces-art/src/phytoplankton.F90:97:37:

   97 |       call self%get_parameter(self%bp, 'bp', '-', 'Temperature sensitivity of growth',default=1.066_rk)
      |                                     1
Error: ‘bp’ at (1) is not a member of the ‘type_base_model’ structure
/home/arthur/Desktop/CODES/SEAMLESS/fabm-pisces-art/src/phytoplankton.F90:98:160:

   98 |       call self%get_parameter(self%logbp, 'logbp', '-', 'Temperature sensitivity of growth (log rate, overwrites bp if given explicitely)', default= log(self%bp))
      |                                                                                                                                                                1
Error: ‘bp’ at (1) is not a member of the ‘type_base_model’ structure
make[2]: *** [extern/fabm/models/pisces/src/CMakeFiles/fabm_models_pisces.dir/build.make:89: extern/fabm/models/pisces/src/CMakeFiles/fabm_models_pisces.dir/phytoplankton.F90.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:755: extern/fabm/models/pisces/src/CMakeFiles/fabm_models_pisces.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

--- a/src/phytoplankton.F90
+++ b/src/phytoplankton.F90
@@ -92,7 +92,6 @@ contains
       call self%get_parameter(self%diatom, 'diatom', '', 'use silicate', default=.false.)
       call self%get_parameter(self%calcify, 'calcify', '', 'calcify', default=.false.)
       call self%get_parameter(self%mumax0, 'mumax0', 'd-1', 'maximum growth rate at 0 degrees Celsius', default=0.8_rk)    ! default=0.8 in NEMO-PISCES 4 code, confirmed by Olivier Aumont 2021-04-21
-      call self%get_parameter(self%logbp, 'logbp', '-', '(log of) Temperature sensitivity of growth', default=0.063913_rk) ! AC -07.12.2021 - Aumont et al. 2015, Table 1, Eq. 4a, Using log(b_p) instead of b_p.
       call self%get_parameter(self%bp, 'bp', '-', 'Temperature sensitivity of growth',default=1.066_rk)
       call self%get_parameter(self%logbp, 'logbp', '-', 'Temperature sensitivity of growth (log rate, overwrites bp if given explicitely)', default= log(self%bp))
       call self%get_parameter(self%fpday, 'fpday', '-', 'day-length factor for growth', default=1.5_rk) ! AC -07.12.2021 - Aumont et al. 2015, Eq 3a

--- a/src/phytoplankton.F90
+++ b/src/phytoplankton.F90
@@ -28,6 +28,7 @@ module pisces_phytoplankton
       logical :: diatom
       logical :: calcify
       real(rk) :: mumax0
+      real(rk) :: bp
       real(rk) :: logbp
       real(rk) :: fpday
       real(rk) :: bresp
@@ -91,7 +92,9 @@ contains
       call self%get_parameter(self%diatom, 'diatom', '', 'use silicate', default=.false.)
       call self%get_parameter(self%calcify, 'calcify', '', 'calcify', default=.false.)
       call self%get_parameter(self%mumax0, 'mumax0', 'd-1', 'maximum growth rate at 0 degrees Celsius', default=0.8_rk)    ! default=0.8 in NEMO-PISCES 4 code, confirmed by Olivier Aumont 2021-04-21
-      call self%get_parameter(self%logbp, 'logbp', '-', '(log of) Temperature sensitivity of growth', default=0.063913_rk) ! AC -07.12.2021 - Aumont et al. 2015, Table 1, Eq. 4a, Using log(b_p) instead of b_p. 
+      call self%get_parameter(self%logbp, 'logbp', '-', '(log of) Temperature sensitivity of growth', default=0.063913_rk) ! AC -07.12.2021 - Aumont et al. 2015, Table 1, Eq. 4a, Using log(b_p) instead of b_p.
+      call self%get_parameter(self%bp, 'bp', '-', 'Temperature sensitivity of growth',default=1.066_rk)
+      call self%get_parameter(self%logbp, 'logbp', '-', 'Temperature sensitivity of growth (log rate, overwrites bp if given explicitely)', default= log(self%bp))
       call self%get_parameter(self%fpday, 'fpday', '-', 'day-length factor for growth', default=1.5_rk) ! AC -07.12.2021 - Aumont et al. 2015, Eq 3a
       call self%get_parameter(self%bresp, 'bresp', 'd-1', 'basal respiration', default=0.033_rk)
       call self%get_parameter(self%pislope_s, 'pislope_s', 'g C (g Chl)-1 (W m-2)-1 d-1', 'P-I slope for small cells', default=2._rk)
@@ -321,7 +324,7 @@ contains
          ! ======================================================================================
          ! Jorn: From p4zint
          !tgfunc = EXP( 0.063913 * tem )  ! Jorn: Eq 4a in PISCES-v2 paper, NB EXP(0.063913) = 1.066 = b_P
-         tgfunc = EXP( self%logbp * tem )  ! AC 07.12.2021 - replaced hard-coded value with parameter for bp. Log(bp) considered instead of bp for perturbation purposes).
+         tgfunc = EXP( self%logbp * tem ) ! AC 07.12.2021 - replaced hard-coded value with parameter for bp. Log(bp) considered instead of bp for perturbation purposes).
          ! ======================================================================================
 
          ! ======================================================================================

--- a/src/phytoplankton.F90
+++ b/src/phytoplankton.F90
@@ -28,7 +28,6 @@ module pisces_phytoplankton
       logical :: diatom
       logical :: calcify
       real(rk) :: mumax0
-      real(rk) :: bp
       real(rk) :: logbp
       real(rk) :: fpday
       real(rk) :: bresp
@@ -87,13 +86,15 @@ contains
       class (type_par),                      pointer :: par_model
       class (type_silicate_half_saturation), pointer :: silicate_half_saturation
 
+      real(rk) :: bp
+      
       allocate(par_model)
 
       call self%get_parameter(self%diatom, 'diatom', '', 'use silicate', default=.false.)
       call self%get_parameter(self%calcify, 'calcify', '', 'calcify', default=.false.)
       call self%get_parameter(self%mumax0, 'mumax0', 'd-1', 'maximum growth rate at 0 degrees Celsius', default=0.8_rk)    ! default=0.8 in NEMO-PISCES 4 code, confirmed by Olivier Aumont 2021-04-21
-      call self%get_parameter(self%bp, 'bp', '-', 'Temperature sensitivity of growth',default=1.066_rk)
-      call self%get_parameter(self%logbp, 'logbp', '-', 'Temperature sensitivity of growth (log rate, overwrites bp if given explicitely)', default= log(self%bp))
+      call self%get_parameter(bp, 'bp', '-', 'Temperature sensitivity of growth',default=1.066_rk)
+      call self%get_parameter(self%logbp, 'logbp', '-', 'Temperature sensitivity of growth (log rate, overwrites bp if given explicitely)', default= log(bp))
       call self%get_parameter(self%fpday, 'fpday', '-', 'day-length factor for growth', default=1.5_rk) ! AC -07.12.2021 - Aumont et al. 2015, Eq 3a
       call self%get_parameter(self%bresp, 'bresp', 'd-1', 'basal respiration', default=0.033_rk)
       call self%get_parameter(self%pislope_s, 'pislope_s', 'g C (g Chl)-1 (W m-2)-1 d-1', 'P-I slope for small cells', default=2._rk)

--- a/src/zooplankton.F90
+++ b/src/zooplankton.F90
@@ -19,7 +19,7 @@ module pisces_zooplankton
       type (type_dependency_id)     :: id_tem, id_nitrfac, id_quotan, id_quotad, id_xfracal, id_wspoc, id_wsgoc
       type (type_diagnostic_variable_id) :: id_zfezoo, id_zgrazing, id_zfrac, id_pcal
 
-      real(rk) :: grazrat, bz, logbz, resrat, xkmort, mzrat, xthresh, xkgraz, ferat, epsher, epshermin, unass, sigma, part, grazflux
+      real(rk) :: grazrat, logbz, resrat, xkmort, mzrat, xthresh, xkgraz, ferat, epsher, epshermin, unass, sigma, part, grazflux
       real(rk) :: xthreshdia, xthreshphy, xthreshzoo, xthreshpoc
       real(rk) :: xprefn, xprefz, xprefd, xprefc, xsizedia, xdismort, phlim
    contains
@@ -33,13 +33,15 @@ contains
       class (type_pisces_zooplankton), intent(inout), target :: self
       integer,                         intent(in)            :: configunit
 
+      real(rk) :: bz
+      
       call self%get_parameter(self%epsher, 'epsher', '1', 'maximum growth efficiency', minimum=0._rk, maximum=1._rk)
       call self%get_parameter(self%epshermin, 'epshermin', '1', 'minimum growth efficiency', minimum=0._rk, maximum=1._rk)
       call self%get_parameter(self%unass, 'unass', '1', 'non-assimilated fraction', default=0.3_rk, minimum=0._rk, maximum=1._rk)
       call self%get_parameter(self%sigma, 'sigma', '1', 'excretion as dissolved organic matter', default=0.6_rk, minimum=0._rk, maximum=1._rk)
       call self%get_parameter(self%grazrat, 'grazrat', 'd-1', 'maximum grazing rate', minimum=0._rk)
-      call self%get_parameter(self%bz, 'bz', '-', 'Temperature sensitivity of grazing rate',default=1.079_rk)
-      call self%get_parameter(self%logbz, 'logbz', '-', 'Temperature sensitivity of grazing rate (log rate, overwrites bz if given explicitely)', default= log(self%bz))
+      call self%get_parameter(bz, 'bz', '-', 'Temperature sensitivity of grazing rate',default=1.079_rk)
+      call self%get_parameter(self%logbz, 'logbz', '-', 'Temperature sensitivity of grazing rate (log rate, overwrites bz if given explicitely)', default= log(bz))
       call self%get_parameter(self%grazflux, 'grazflux', '(m mol C L-1)-1', 'flux-feeding rate', minimum=0._rk)
       call self%get_parameter(self%xkgraz, 'xkgraz', 'mol C L-1', 'half-saturation constant for grazing', default=20.e-6_rk, minimum=0._rk)
       call self%get_parameter(self%xprefn, 'xprefn', '-', 'preference for nanophytoplankton', minimum=0._rk)

--- a/src/zooplankton.F90
+++ b/src/zooplankton.F90
@@ -19,7 +19,7 @@ module pisces_zooplankton
       type (type_dependency_id)     :: id_tem, id_nitrfac, id_quotan, id_quotad, id_xfracal, id_wspoc, id_wsgoc
       type (type_diagnostic_variable_id) :: id_zfezoo, id_zgrazing, id_zfrac, id_pcal
 
-      real(rk) :: grazrat, logbz, resrat, xkmort, mzrat, xthresh, xkgraz, ferat, epsher, epshermin, unass, sigma, part, grazflux
+      real(rk) :: grazrat, bz, logbz, resrat, xkmort, mzrat, xthresh, xkgraz, ferat, epsher, epshermin, unass, sigma, part, grazflux
       real(rk) :: xthreshdia, xthreshphy, xthreshzoo, xthreshpoc
       real(rk) :: xprefn, xprefz, xprefd, xprefc, xsizedia, xdismort, phlim
    contains
@@ -38,7 +38,8 @@ contains
       call self%get_parameter(self%unass, 'unass', '1', 'non-assimilated fraction', default=0.3_rk, minimum=0._rk, maximum=1._rk)
       call self%get_parameter(self%sigma, 'sigma', '1', 'excretion as dissolved organic matter', default=0.6_rk, minimum=0._rk, maximum=1._rk)
       call self%get_parameter(self%grazrat, 'grazrat', 'd-1', 'maximum grazing rate', minimum=0._rk)
-      call self%get_parameter(self%logbz, 'logbz', '-', '(log of) Temperature sensitivity of grazing rate',default=0.07608_rk) ! AC 07.12.2021 
+      call self%get_parameter(self%bz, 'bz', '-', 'Temperature sensitivity of grazing rate',default=1.079_rk)
+      call self%get_parameter(self%logbz, 'logbz', '-', 'Temperature sensitivity of grazing rate (log rate, overwrites bz if given explicitely)', default= log(self%bz))
       call self%get_parameter(self%grazflux, 'grazflux', '(m mol C L-1)-1', 'flux-feeding rate', minimum=0._rk)
       call self%get_parameter(self%xkgraz, 'xkgraz', 'mol C L-1', 'half-saturation constant for grazing', default=20.e-6_rk, minimum=0._rk)
       call self%get_parameter(self%xprefn, 'xprefn', '-', 'preference for nanophytoplankton', minimum=0._rk)

